io/jvm/src/main/scala/scalaz/effect/RTS.scala

```diff
@@ -557,16 +540,16 @@ private object RTS {
                     } finally enterAsyncEnd()
 
                   case IO.Tags.Attempt =>
-                    val io = curIo.asInstanceOf[IO.Attempt[E, Any, Any]]
+                    val io = curIo.asInstanceOf[IO.Attempt[E, Any, Any, Any]]
 
                     curIo = io.value
 
-                    stack.push(Catcher)
+                    stack.push(io)
```

Comment:
@ktonga 

You can make this faster with: 

```scala
stack.push(io.succ)
```

Then it is 1 less level of wrapping.

Reply:
@ktonga 

Sorry, I was wrong about this and it will cause test failures.

See [this line](
https://github.com/scalaz/scalaz-ioz/blob/4e1c0cc8dcc6766dd165cad2bbee1f218fd238f6/io/jvm/src/main/scala/scalaz/effect/RTS.scala#L253).

We rely on the `Attempt` class to detect that it's an error handler. So we need to push `io` so we can later detect it has an error handler and extract it in `catchError`.